python/docs/basic.rst

```diff
@@ -33,6 +33,24 @@ name of the function being decorated:
        pass
 
 
+start_range and end_range
+-------------------------
+
+In certain situations, it is impossible to use ``annotate()``,
```

Comment:
How does the `start/end` API help with async functions? (not contesting your point, but actually curious)

Reply:
Async functions make no guarantee about when they end when started so it's easy to have a pattern that looks like:

- start A
- start B
- end A
- start D
- end C
- end D
- end B

Which makes using a stack based approach of push/pop impossible.

Reply:
Gotcha -- yeah that makes sense.

Reply:
Added async code. I realized we don't document the `push_range` and `pop_range` APIs because there's no really good reason to use them, as compared to `annotate`.